pkg/operator/ceph/object/rgw.go

```diff
@@ -284,6 +290,11 @@ func (c *clusterConfig) deleteStore() error {
 
 		objContext.Endpoint = c.store.Status.Info["endpoint"]
 
+		err = disableRGWDashboard(objContext)
```

Comment:
Will this disable the dashboard in the case where there are 2 CephObjectStores created and 1 of the CephObjectStores is removed?

Reply:
@BlaineEXE, For the second object store `dashboard-admin` user won't be created. But currently `DeleteUser` will return success if user does not exist, hence `disableRGWDashboard` will go and ahead reset the rgw credentials for the dashboard. Included new check using `GetUser` before calling `DeleteUser` to resolve this issue. Good catch

Reply:
I'm still concerned about this. In my testing, I found that if I created 2 `CephObjectStores`, one named `my-store` and one named `my-store-2`, when I delete `my-store`, I see in the logs the below

```
2020-07-28 18:47:32.073610 D | exec: Running command: radosgw-admin user rm --uid dashboard-admin --rgw-realm=my-store --rgw-zonegroup=my-store --rgw-zone=my-store --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --name=client.admin --keyring=/var/lib/rook/rook-ceph/client.admin.keyring
```

I don't think this should be happening, however, because the admin user still needs to exist for `my-store-2` which still exists. It's possible I'm missing something, but I want to make sure we are testing this scenario and not removing the RGW dashboard unless ALL `CephObjectStores` are deleted

Reply:
@BlaineEXE AFAIK dashboard can be enable for one independent object store, in above case it was enabled for `my-store` and not for `my-store-2`. Still ceph dashboard lists all the daemon(don't know why) but not the buckets or users in the second object store. In the above case when `my-store` was deleted, then dashboard will be disabled which is acceptable IMHO

Reply:
> AFAIK dashboard can be enable for one independent object store

Can you explain more what you mean here? Do you mean that the Ceph Dashboard is only able to interface with one object store at a time (a)? Or do you mean that in Rook the RGW dashboard can be selectively enabled/disabled for each CephObjectStore (b)?

If it's the first thing that I tagged (a) above, I still don't consider it "acceptable," but it does sound like it might be technically necessary at this point. From a user perspective, they will find this behavior unexpected, and I would still consider it a bug, even if it is a bug in Ceph itself. From the huddle discussion, I think I'm okay with this PR getting merged, but before we do so, we should (1) open a new issue to track this in Rook (and in Ceph?) and (2) add to the PendingReleaseNotes **Known issues** section that this will happen if multiple CephObjectStores exist and one of the stores is deleted.

Reply:
@BlaineEXE if I understand correctly issue reside with current ceph dashboard. I will add necessary documentation. Also will the check with ceph dashboard devs whether any open issues are there and how much work need to done for the same.

Reply:
I went through the logs to verify, and it does look like the changes will indeed not delete the rgw dashboard admin user for the first-created store when the second-created store is removed, so I think your fix is good. Would you add a known issue to the release notes and document somewhere in the Rook docs that only the first-created CephObjectStore will have a dashboard?

Reply:
@BlaineEXE Sure will add that and update the PR